src/runDiff.ts

```diff
@@ -0,0 +1,225 @@
+  summary: string;
+}
+
+/**
+ * Executes CDKTF diff command, parses the output, and provides structured results.
+ * Handles different output patterns to determine if there are changes, errors, or no changes.
+ */
+export class RunDiff {
+  octokit: Octokit & Api & { paginate: PaginateInterface };
+  context: Context;
+  inputs: ActionInputs;
+
+  /**
+   * Initialize clients and member variables.
+   */
+  constructor() {
+    // Get inputs first
+    this.inputs = {
+      githubToken: core.getInput("github_token", { required: true }),
+      jobName: core.getInput("job_name", { required: true }),
+      outputFilename: core.getInput("output_filename", { required: true }),
+      ref: core.getInput("ref", { required: true }),
+      stack: core.getInput("stack", { required: true }),
+      stubOutputFile: core.getInput("stub_output_file"),
+      terraformVersion: core.getInput("terraform_version") || "1.8.0",
+      workingDirectory: core.getInput("working_directory") || "./",
+      skipSynth: core.getBooleanInput("skip_synth"),
+      artifactName: core.getInput("artifact_name"),
+    };
+
+    // Then create octokit with the token
+    this.octokit = github.getOctokit(this.inputs.githubToken);
+    this.context = github.context;
+    core.debug("Context:");
+    core.debug(JSON.stringify(this.context));
+  }
+
+  /**
+   * Main execution method. Gets the job ID, runs the CDKTF diff, and processes the results.
+   * Sets action outputs and writes results to a file. Fails the action if errors are detected.
+   */
+  async run() {
+    // Get job information
+    const { job_id, html_url } = await this.getJobId();
+
+    // Run the diff
+    const { result_code, summary } = await this.runDiff();
+
+    // Prepare outputs
+    const outputs: ActionOutputs = {
+      html_url,
+      job_id: job_id.toString(),
+      result_code,
+      stack: this.inputs.stack,
+      summary
+    };
+
+    // Set action outputs
+    Object.entries(outputs).forEach(([key, value]) => {
+      core.setOutput(key, value);
+    });
+
+    // Write outputs to file
+    const outputPath = path.join(this.inputs.workingDirectory, this.inputs.outputFilename);
+    fs.writeFileSync(outputPath, JSON.stringify(outputs));
+
+    if(result_code === "1") {
+      core.setFailed(summary);
+    }
+  }
+
+  /**
+   * Retrieves the job ID and HTML URL for the current workflow job.
+   * 
+   * @returns Promise containing the job ID and HTML URL
+   * @throws Error if the job cannot be found
+   */
+  async getJobId(): Promise<{ job_id: number; html_url: string }> {
+    const octokitPaginatedJobs = await this.octokit.paginate(
+      this.octokit.rest.actions.listJobsForWorkflowRun,
+      {
+        ...github.context.repo,
+        run_id: github.context.runId
+      }
+    );
+
+    const job = octokitPaginatedJobs.find(j => j.name === this.inputs.jobName);
+
+    if (job) {
+      return {
+        job_id: job.id,
+        html_url: job.html_url || ""
```

Comment:
Personally, I would love to see this follow [TS/JS naming conventions](https://google.github.io/styleguide/tsguide.html#camel-case):
```suggestion
        jobId: job.id,
        htmlUrl: job.html_url || ""
```

Reply:
I agree 100%, unfortunately this would introduce a breaking change. We can discuss updating this in a later PR, for now this PR is just migrating the existing functionality into typescript. We have stories on our roadmap for improvements

Reply:
So, I see this reasoning for the `outputs` object on line 106 as it's used by an outside process, but it doesn't seem like it applies here. This appears to be a private function that could declare properties as it wants. Either way, it doesn't seem like a huge issue.